app/classifier/services.py

```diff
@@ -53,13 +54,18 @@ def train(self, dataloaders, criterion, optimizer, scheduler):
             if phase == 'training':
               loss.backward()
               optimizer.step()
-          # statistics
+          # statistics  
           running_loss += loss.item() * inputs[0].size(0)
           running_corrects += torch.sum(preds == labels.data)
         if phase == 'training' and scheduler is not None:
           scheduler.step()
         epoch_loss = running_loss / len(dataloaders[phase].dataset)
-        epoch_acc = running_corrects.double() / len(dataloaders[phase].dataset)
+        epoch_acc = f1_score(labels.tolist(), preds.tolist(), average="macro")
```

Comment:
Big mistake here, comparing predictions of batch with labels of whole epoch (much larger). Must compare epoch-to-epoch, else it will not be a realistic measure.